src/main/java/nextstep/app/ui/MemberController.java

```diff
@@ -24,6 +27,19 @@ public ResponseEntity<List<Member>> list() {
         return ResponseEntity.ok(members);
     }
 
+    @GetMapping("/members/me")
+    public ResponseEntity<Member> getMember() {
+        SecurityContext context = SecurityContextHolder.getContext();
+        String email = (String) context.getAuthentication().getPrincipal();
+
+        Optional<Member> memberOpt = memberRepository.findByEmail(email);
+        if (memberOpt.isEmpty()) {
+            throw new IllegalStateException("회원을 찾을 수 없습니다. (email=%s)".formatted(email));
+        }
+
+        return ResponseEntity.ok(memberOpt.get());
+    }
```

Comment:
orElseThrow()를 사용하면 더 간결하고 처리할 수 있겠네요 😄 
```suggestion
        Member member = memberRepository.findByEmail(email)
                .orElseThrow(() -> new IllegalStateException("회원을 찾을 수 없습니다. (email=%s)".formatted(email)));

        return ResponseEntity.ok(member);
    }
```

Reply:
[dcc8acc](https://github.com/next-step/spring-security-authorization/pull/27/commits/dcc8acc913ffd343d8ef45a7932d93f6ba8c9240) 커밋에 해당 내용 반영해봤습니다 :)

---

별개로 조금 다른 질문이긴 하지만, 다음 두 가지 내용이 궁금해서 질문드려봅니다..!

1. _위 `MemberController::getMember` 로직에서 `IllegalStateException`과 `IllegalArgumentException` 중 무엇이 더 적절할까요?_

두 예외 중 무엇을 발생시킬 지 고민했는데, 이미 인증 필터에서 인증된 상태의 유저를 찾았고 이 상태에서 username과 일치하는 member가 없다는 것은 클라이언트의 잘못(IllegalArgumentException)이라기 보다는 서버 상태의 이상을 의미하기 때문에 IllegalStateException으로 선언했습니다. 리뷰어님이라면 어떤 예외를 사용하셨을지 궁금합니다..!

2. _컨트롤러에서 SecurityContext를 직접 접근하는게 괜찮을까요?_

컨트롤러는 Spring MVC의 영역이고, 인증이 다 끝난 상태에서 수행되는 비즈니스 로직인데 여기서 SecurityContext를 접근해도 괜찮은 걸까라는 고민이 되더군요. 또, 실무 코드라면 memberRepository.findByEmail등의 로직을 매번 작성해야하는 것이 번거로워 `@AuthenticatedMember`라는 애노테이션을 만들고 MethodArgumentResolver로 Member객체를 주입해줄 것 같긴 한데 여기에 대한 리뷰어님의 의견이 궁금합니다 :-)

Reply:
> 1. 위 MemberController::getMember 로직에서 IllegalStateException과 IllegalArgumentException 중 무엇이 더 적절할까요?

A. email이 조회되지 않는다면, 말씀하신 것 처럼 애플리케이션의 상태가 예상과 다르게 흘러갔다고 볼 수 있을 것 같아요.
저도 선호님 처럼 IllegalStateException 이 더 적절하다고 생각합니다.
 다만 실무에서는 CustomException을 정의하여 보다 명확한 예외를 던지는 방식을 선택할 것 같네요.  
 CustomException을 통해 예외의 의미를 명확하게 전달하고 일관된 방식으로 예외를 처리하는 방식을 선호합니다.😄 

> 2. 컨트롤러에서 SecurityContext를 직접 접근하는 것이 적절한가?

A.저도 실무에서는 선호님께서 말씀해 주신 것처럼 커스텀 애노테이션을 만들어 사용하는 방식을 선호합니다. 😄
컨트롤러는 보안 관련 로직을 다루기보다는, 요청을 처리하는 역할에 집중하는 것이 좋다고 생각해요.
또한, 인증 방식이 변경될 때 컨트롤러를 수정할 필요 없이 ArgumentResolver만 변경하면 되므로, 유지보수적인 측면에서도 더 좋은 방식이라고 생각합니다. 👍